cmd/osv-scanner/scan/main.go

```diff
@@ -221,6 +229,18 @@ func action(context *cli.Context, stdout, stderr io.Writer) (reporter.Reporter,
 		callAnalysisStates = createCallAnalysisStates(context.StringSlice("call-analysis"), context.StringSlice("no-call-analysis"))
 	}
 
+	offlineVulns := context.Bool("experimental-offline-vulnerabilities")
+	disableTransitive := context.Bool("experimental-no-resolve")
+	if context.Bool("experimental-offline") {
+		// Set the corresponding offline flags, unless they were already set explicitly.
+		if !context.IsSet("experimental-offline-vulnerabilities") {
```

Comment:
Why do we need to check if they were already set? if someone is setting `--experimental-offline`, then IMO that should just always disable everything regardless of what other flags are set?

Reply:
I think I may understand the use case a bit here: someone wants to disable all network requests but not the ones for vulnerabilities. 
Also as Michael mentioned below, `CompareOffline` also controls others things for example license scanning, maybe we should change it to a struct with a list of booleans? Each boolean field corresponds to one type of request requiring network.
Even further, can we change `experimental-offline` to be a list of strings taking `all` or `vulnerability,license` and `all` takes precedence over all other.

Reply:
NB: I moved this logic to the `Action` of the offline flag in [aa0a734](https://github.com/google/osv-scanner/pull/1342/commits/aa0a73405e98ae8d7b009e640456e4f571c73d21), but I've not addressed this discussion.

Reply:
IMO let's keep this simple for now and go with @michaelkedar 's approach. 

`--experimental-offline` should just only be set when  somebody wants to turn everything off, and can't be overriden. There's already individual flags to control behaviour (after this PR) for the independent bits. 

Re taking a slice of strings instead, we went with that for `--call-analysis` to take in a list of languages, but I think full list of languages would be much larger than the list of functionality where we can control online/offline. (i.e. we'd have --call-analysis-rust, --call-analysis-go, --call-analysis-python, --call-analysis-{language}, ...)

And since this is all experimental, we can change this later :)